app/page.tsx

```diff
@@ -1,98 +1,77 @@
-// Homepage - User login and signup
 "use client";
-import React, { ChangeEventHandler, FormEventHandler, useState } from "react";
-import { signIn } from "next-auth/react";
+
+import { useEffect, useState, useRef } from "react";
+import { AuthorizeOptions } from "@/lib/zoomapi";
+import { signIn, useSession } from "next-auth/react";
+import { SessionProvider } from "next-auth/react";
+import { ZoomApiWrapper } from "@/lib/zoomapi";
 import { useRouter } from "next/navigation";
+import { log, Action } from "@/lib/log";
 import Alert from "@/components/Alert";
-import { Action, log } from "@/lib/log";
+import { LoadingComponent } from "@/components/LoadingComponent";
 
-const Login = () => {
+function App() {
   const [error, setError] = useState("");
-  const [userInfo, setUserInfo] = useState({
-    email: "",
-    password: "",
-  });
+  const { data: session } = useSession();
+  const effectRan = useRef(false);
+
   const router = useRouter();
-  const { email, password } = userInfo;
 
-  const handleChange: ChangeEventHandler<HTMLInputElement> = ({ target }) => {
-    const { name, value } = target;
-    setUserInfo({ ...userInfo, [name]: value });
-  };
+  async function getZoomApi(): Promise<ZoomApiWrapper> {
+    const zoomModule = await import("@/lib/zoomapi");
+    return zoomModule.zoomApi;
```

Comment:
I think I was expecting this to be something like:
```suggestion
    const zoomImport = process.env.MOCK_ZOOM_API ? import("@/lib/fakezoomapi") : import("@/lib/zoomapi");
    const zoomModule = await zoomImport
    return zoomModule.zoomApi;
```
and then not having to have other logic in the ZoomApi about cypress.

Does this not work?
If it doesn't work, can you explain what the problem is?